tests/dag_processing/test_job_runner.py

```diff
@@ -762,6 +763,95 @@ def test_scan_stale_dags_standalone_mode(self):
             active_dag_count = session.query(func.count(DagModel.dag_id)).filter(DagModel.is_active).scalar()
             assert active_dag_count == 1
 
+    @conf_vars(
+        {
+            ("scheduler", "standalone_dag_processor"): "False",
+        }
+    )
+    def test_scan_stale_dags_when_dag_folder_change(self):
+        """
+        Ensure dags from old dag_folder is marked as stale when dag processor
+         is running as part of scheduler.
+        """
+
+        def get_dag_string(dag_id) -> str:
+            return (
+                f"from __future__ import annotations\n"
+                f"import datetime\n"
+                f"from airflow.models.dag import DAG\n"
+                f"from airflow.operators.empty import EmptyOperator\n"
+                f"with DAG(\n"
+                f'  dag_id="{dag_id}",\n'
+                f"  schedule=datetime.timedelta(hours=4),\n"
+                f"  start_date=datetime.datetime(2021, 1, 1),\n"
+                f"  catchup=False,\n"
+                f") as dag:\n"
+                f'    task1 = EmptyOperator(task_id="task1")\n'
```

Comment:
Can you use a DAG from tests/dags directory instead of this?

Reply:
Sure, I have updated it. PTAL